src/base/IoManip.h

```diff
@@ -109,17 +108,17 @@ class AsList
 public:
     explicit AsList(const Container &c): container(c) {}
 
-    /// a character to print before the first item (if any)
-    auto &prefixedBy(const char ch) { prefix = ch; return *this; }
+    /// a c-string to print before the first item (if any). Caller must ensure lifetime.
```

Comment:
FWIW, I have considered using SBuf to eliminate these lifetime concerns, but decided that we should not do it (yet) since existing callers do not need this, and there is a chance that no future caller will need this.

Reply:
Again, thought. In practice, it's highly likely that the separators will be literals, so no lifetime concerns.
OTOH, bringing SBuf in would mean that all the required machinery would also need to be pulled in.
FWIW, I considered using std::string, but it also felt overkill for this specific purpose

Reply:
std::string may be more efficient long-term with the C++17 `""s` syntax constants. But that can be an optimization issue for later.